### PR TITLE
track buffer voided promises in plugin server

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Adding some important instrumentation

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: plugin-server/src/worker/vm/utils.ts [hunk 1]
```diff
@@ -72,3 +73,66 @@ export const addPublicJobIfNotExists = async (
...
     )
 }
+
+type BufferOptions = {
+    limit: number
+    timeoutSeconds: number
+    onFlush: (objects: any[], points: number) => void | Promise<void>
+}
+
+export function createBuffer(opts: Partial<BufferOptions>, statsd?: StatsD) {
```
Comment (yakkomajuri) [inline plugin-server/src/worker/vm/utils.ts:83]: this is essentially a copy 🍝 from https://github.com/PostHog/plugin-contrib/blob/main/src/buffer.ts with instrumentation added

File: plugin-server/src/worker/vm/utils.ts [hunk 2]
```diff
@@ -72,3 +73,66 @@ export const addPublicJobIfNotExists = async (
...
+                buffer.triggerFlushInstrumented()
+            } else if (!buffer._timeout) {
+                // if not, make sure there's a flush timeout
+                buffer._timeout = setTimeout(
+                    () => buffer.triggerFlushInstrumented(),
+                    buffer._options.timeoutSeconds * 1000
+                )
+            }
+        },
+        triggerFlushInstrumented: () => {
```
Comment (yakkomajuri) [inline plugin-server/src/worker/vm/utils.ts:115]: this is the key here

**Review by fuziontech — APPROVED** [review]
looks reasonable to me

File: plugin-server/src/worker/vm/utils.ts (hunk 1 above)
Comment (fuziontech) [reply plugin-server/src/worker/vm/utils.ts:83]: checks out

**Pull request merged**